### PR TITLE
Soca increment plots using eva

### LINKED ISSUE #342: Produce SOCA increment plots using EVA
Essentially having SOCA increment plots since `eva_increments` task is already created, by @asewnath 😄. The only difference with the `geos_atmosphere` would be the `SOCArestart` data reader in EVA looks for a static file (`soca_gridspec.nc`) and that is located in the `cycle_dir` under `soca/`

We could have the surface slices for `Temp` and `Salt` which are 3D. `ave_ssh` is in 2D so that wouldn't need slicing.

### PR DESCRIPTION
**Pull request by asewnath**
## Description

Adding ability to produce increment plots for SOCA in 3dvar suite. Since the eva config file is purely an example, I'd expect that it would be updated to include more diagnostics before this PR is merged.

Resolves #342

### CODE REVIEW

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            variable: experiment_increment::SOCAgrid::lon
+          latitude:
+            variable: experiment_increment::SOCAgrid::lat
+          data:
+            variable: experiment_increment::SOCAVars::ave_ssh
+          label: ave_ssh increment
+          colorbar: true
+          cmap: 'bwr'
+          vmin: -100
+          vmax: 100
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: ranges should be: -2, +2

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      layout: [1,1]
+      title: 'SOCA Increment'
+      output name: '{{cycle_dir}}/eva/increment/map_plots/${variable}/inc_${variable}.png'
+    plots:
+      - mapping:
+          projection: plcarr
+          domain: global
+        add_map_features: ['coastline']
+        add_colorbar:
+          label: Surface Pressure Increment
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: SSH increment

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        add_grid:
+        layers:
+        - type: MapGridded
+          longitude:
+            variable: experiment_increment::SOCAgrid::lon
+          latitude:
+            variable: experiment_increment::SOCAgrid::lat
+          data:
+            variable: experiment_increment::SOCAVars::Temp
+            slices: '[20,...]'
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: slice should be at 0, the top layer.

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      layout: [1,1]
+      title: 'Soca Increment'
+      output name: '{{cycle_dir}}/eva/increment/map_plots/${variable}/inc_${variable}.png'
+    plots:
+      - mapping:
+          projection: plcarr
+          domain: global
+        add_map_features: ['coastline']
+        add_colorbar:
+          label: Temperature Increment
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: If slice is 0, it should be SST increment

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          latitude:
+            variable: experiment_increment::SOCAgrid::lat
+          data:
+            variable: experiment_increment::SOCAVars::Temp
+            slices: '[20,...]'
+          label: Temp increment
+          colorbar: true
+          cmap: 'bwr'
+          vmin: -1
+          vmax: 1
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: range: -3, 3

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        add_grid:
+        layers:
+        - type: MapGridded
+          longitude:
+            variable: experiment_increment::SOCAgrid::lon
+          latitude:
+            variable: experiment_increment::SOCAgrid::lat
+          data:
+            variable: experiment_increment::SOCAVars::Salt
+            slices: '[20,...]'
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: slice, 0, Sea Surface Salinity (SSS)

File: src/swell/suites/3dvar/eva/increment-geos_ocean.yaml [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      layout: [1,1]
+      title: 'Soca Increment'
+      output name: '{{cycle_dir}}/eva/increment/map_plots/${variable}/inc_${variable}.png'
+    plots:
+      - mapping:
+          projection: plcarr
+          domain: global
+        add_map_features: ['coastline']
+        add_colorbar:
+          label: Temperature Increment
```
Comment (Dooruk) [inline src/swell/suites/3dvar/eva/increment-geos_ocean.yaml]: SSS Increment

**Review by Dooruk — COMMENTED** [review]
@asewnath, thanks again for working on this.

I have a branch with the CICE6 sea-ice analysis and that will require a few more additional steps to this, regarding map domain. I can merge this to that branch and work on the comments I just made to this PR, or we can just merge this one first if you want to fix the code. Let me know.

**Comment by asewnath** [comment]
> @asewnath, thanks again for working on this.
> 
> I have a branch with the CICE6 sea-ice analysis and that will require a few more additional steps to this, regarding map domain. I can merge this to that branch and work on the comments I just made to this PR, or we can just merge this one first if you want to fix the code. Let me know.

Please feel free to make the config yaml changes yourself

**Review by Dooruk — APPROVED** [review]
Thanks, all good!

**Pull request merged**